src/main/java/de/unijena/cheminf/mortar/gui/controls/ToggleSwitch.java

```diff
@@ -0,0 +1,196 @@
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package de.unijena.cheminf.mortar.gui.controls;
+
+import javafx.animation.FillTransition;
+import javafx.animation.ParallelTransition;
+import javafx.animation.TranslateTransition;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.scene.control.Control;
+import javafx.scene.effect.DropShadow;
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Circle;
+import javafx.scene.shape.Rectangle;
+import javafx.util.Duration;
+
+/**
+ * A toggle switch to en- and disable features in settings.
+ *
+ * @author Zeynep Dagtekin
+ * @version 1.0.0.0
+ */
+public class ToggleSwitch extends Control {
+    //<editor-fold desc="private final class constants" defaultstate="collapsed">
+    /**
+     * Button.
+     */
+    private final Circle switchButton;
+    /**
+     * Background of the switch.
+     */
+    private final Rectangle switchBackground;
+    /**
+     * Boolean property to keep track of the state.
+     */
+    private final SimpleBooleanProperty switchedOn;
+    /**
+     * transition of the Circle from one side to the other.
+     */
+    private final TranslateTransition switchAnimation;
+    /**
+     * Color transition.
+     */
+    private final FillTransition fillAnimation;
+    /**
+     * Parallel transition of color and Circle.
+     */
+    private final ParallelTransition switchTransition;
+    //</editor-fold>
+    /**
+     * Constructor.
+     */
+    public ToggleSwitch() {
+        /**
+         * code inspired by "JavaFX UI: iOS Style Toggle Switch", uploaded by Almas Baimagambetov on YouTube
+         * https://youtu.be/maX5ymmQixM?si=v2ULa57-pjCmoQlf, 05/17/2024, 10:33
+         */
+        super();
+        this.switchedOn = new SimpleBooleanProperty(false);
+        this.switchBackground = new Rectangle(45, 18);
+        this.switchBackground.setArcWidth(18);
+        this.switchBackground.setArcHeight(18);
+        this.switchBackground.setLayoutX(-50);
+        this.switchBackground.setFill(Color.LIGHTGRAY);
+        this.switchBackground.setStroke(Color.DARKGRAY);
+        this.switchButton = new Circle(10);
+        this.switchButton.setCenterX(-40);
+        this.switchButton.setCenterY(9);
+        this.switchButton.setFill(Color.WHITE);
+        this.switchButton.setStroke(Color.DARKGRAY);
+        this.switchButton.setEffect(new DropShadow(5, Color.GRAY));
+        this.switchAnimation = new TranslateTransition(Duration.seconds(0.25));
+        this.switchAnimation.setNode(this.switchButton);
+        this.fillAnimation = new FillTransition(Duration.seconds(0.25));
```

Comment:
Size parameters like the rectangle size, the circle size, etc., should be configurable via setters (getters are then also needed, of course). Also the colors and transition times. The values you are using here should be defined in class constants and used as defaults. Shadows should be able to be turned off and on.

*If* we would do this more professionally, we would also define resizing and how the switch behaves for a given min/preferred/max height/width. But this goes beyond our current use case. 
But at least, that these behaviours are not implemented should be noted in the class documentation because e.g. the setPrefHeight() method is defined by the extended Control class but your switch behaves weirdly if it is used.